src/OpenTelemetry/Instrumentation/InstrumentationEventSource.cs

```diff
@@ -32,7 +32,7 @@ public class InstrumentationEventSource : EventSource
         [NonEvent]
         public void ExceptionInCustomSampler(Exception ex)
         {
-            if (this.IsEnabled(EventLevel.Warning, EventKeywords.All))
+            if (this.IsEnabled(EventLevel.Warning, (EventKeywords)(-1)))
```

Comment:
nit: having a constant, eg.: `EventKeyworsAll`, with a comment about net452 will make clear why `EventKeywords.All` is not used.

Reply:
This might be an overkill? `(EventKeywords)(-1)` already exists in the current code base. And it is scattered in several projects. Defining the same constant in many projects might be confusing? (on the other side, putting a const in the API package and reference it in other packages seems too much for this issue? I've also struggled with the idea of having an `extension` but gave up 😄)

Another approach is to have #if/#else/#endif based on `NET452`. What's your persepctive?

Reply:
ah, I wasn't aware that it was used in many places. My personal preference will be to clear it up, but it is a lower priority issue. Anyway, if we ever do it it should be in a dedicated PR since it is going to touch many files. I would rather not add more #if to the code, keep as it is.